airbyte-cdk/python/connector_builder/message_grouper.py

```diff
@@ -0,0 +1,192 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import json
+from json import JSONDecodeError
+from typing import Any, Iterable, Iterator, Mapping, Optional, Union
+from urllib.parse import parse_qs, urlparse
+
+from airbyte_protocol.models.airbyte_protocol import ConfiguredAirbyteCatalog
+
+from airbyte_cdk.models import AirbyteLogMessage, AirbyteMessage, Type
+from airbyte_cdk.sources.declarative.declarative_source import DeclarativeSource
+from airbyte_cdk.utils.schema_inferrer import SchemaInferrer
+from connector_builder.models import StreamRead, StreamReadPages, HttpResponse, HttpRequest, StreamReadSlices
+import logging
+from copy import deepcopy
+
+
+class MessageGrouper:
```

Comment:
bulk of the logic is pasted from https://github.com/airbytehq/airbyte-platform-internal/blob/master/oss/airbyte-connector-builder-server/connector_builder/impl/default_api.py